modules/node/src/appRegistry/appRegistry.service.ts

```diff
@@ -289,6 +319,146 @@ export class AppRegistryService implements OnModuleInit {
     logger.log(`Exchange amounts are within ${ALLOWED_DISCREPANCY_PCT}% of our rate ${ourRate}`);
   }
 
+  // TODO most of the code is duplicated from validateSimpleLinkedTransfer
+  // should refactor this into single validation method based on condition type
```

Comment:
We could have a common `validateTransfer` that encapsulates all the validation general transfer apps need.